app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt

```diff
@@ -449,6 +458,27 @@ class WireNotificationManager @Inject constructor(
         }
     }
 
+    private fun verifyIfShouldPlayPingSound(
+        currentScreen: CurrentScreen,
+        notifications: List<LocalNotificationConversation>
+    ) {
+        if (currentScreen is CurrentScreen.Conversation) {
+            val conversationId = currentScreen.id
+            val containsPingMessage = notifications
+                .filter { it.id == conversationId }
+                .any {
+                    it.messages.any { message -> message is LocalNotificationMessage.Knock }
+                }
```

Comment:
```suggestion
                .any {
                    it.id == conversationId &&
                    it.messages.any { message -> message is LocalNotificationMessage.Knock }
                }
```

TINY performance boost without losing readability imho.
No need to iterate through all the items and then iterate again on the filtered ones.
Just iterate once and that's it.